numpy/_core/tests/test_multiarray.py

```diff
@@ -266,6 +266,52 @@ def test_void_align(self):
         a = np.zeros(4, dtype=np.dtype([("a", "i4"), ("b", "i4")]))
         assert_(a.flags.aligned)
 
+    def test_xcontiguous_load_txt(self):
+        d = '1.0 '
+        d_short = d * 5
+        d_long = d * (1 << 16)
+
+        s = io.StringIO(d_short)
+        a = np.loadtxt(s)
+        assert_(a.flags.c_contiguous)
+        assert_(a.flags.f_contiguous)
+
+        s = io.StringIO(d_long)
+        a = np.loadtxt(s)
+        assert_(a.flags.c_contiguous)
+        assert_(a.flags.f_contiguous)
+
+        s = io.StringIO('\n'.join([d_short] * 5))
+        a = np.loadtxt(s)
+        assert_(a.flags.c_contiguous)
+        assert_(a.flags.f_contiguous is False)
+
+        s = io.StringIO('\n'.join([d_long] * 5))
+        a = np.loadtxt(s)
+        assert_(a.flags.c_contiguous)
+        assert_(a.flags.f_contiguous is False)
+
+        s = io.StringIO(d_short)
+        a = np.loadtxt(s, ndmin=2)
+        assert_(a.flags.c_contiguous)
+        assert_(a.flags.f_contiguous)
+
+        s = io.StringIO(d_long)
+        a = np.loadtxt(s, ndmin=2)
+        assert_(a.flags.c_contiguous)
+        assert_(a.flags.f_contiguous)
+
+        s = io.StringIO('\n'.join([d_short] * 5))
+        a = np.loadtxt(s, ndmin=2)
+        assert_(a.flags.c_contiguous)
+        assert_(a.flags.f_contiguous is False)
+
+        s = io.StringIO('\n'.join([d_long] * 5))
+        a = np.loadtxt(s, ndmin=2)
+        assert_(a.flags.c_contiguous)
+        assert_(a.flags.f_contiguous is False)
```

Comment:
If you are up to it, one thing that would be cute, is to write this as a `pytest.mark.parametrize` pattern, like:
```
@pytest.mark.parametrize("row", ["1.0 " * 5, "1.0 " * (1 << 16)])
@pytest.mark.parametrize("ndmin", [0, 2])
def test_xcontiguous_load_txt(self, row, ndmin):
```
To make the test much less repetitive.

As an aside: using `assert_()` is an ancient left-over pytest prefers `assert a.flags.c_contiguous`.  This file uses it a lot, so it's OK either way.

Reply:
Fine, but pytest may fail if the parameter is too long, so `row_size` maybe a better option.